solver/src/encoding.rs

```diff
@@ -0,0 +1,84 @@
+use model::{OrderKind, UserOrder};
+use primitive_types::U256;
+
+const TRADE_SIZE: usize = 204;
+
+/// Creates the data which the smart contract's `decodeTrade` expects.
+pub fn encode_trade(
```

Comment:
probably doesn't make a difference, but curious why not have this as an extension on the Order struct directly?

Reply:
Not sure yet where it should live and a function is easier to move. Maybe we'll have a Trade struct later on which it can be implemented. And this function is only relevant to the solver and not the orderbook.